Result/ResultType.swift

```diff
@@ -65,9 +71,9 @@ public extension ResultType {
 }
 
 /// Protocol used to constrain `tryMap` to `Result`s with compatible `Error`s.
-public protocol ErrorTypeConvertible: ErrorType {
-	typealias ConvertibleType = Self
-	static func errorFromErrorType(error: ErrorType) -> ConvertibleType
+public protocol ErrorTypeConvertible: ResultErrorType {
+	associatedtype ConvertibleType = Self
```

Comment:
@ikesyo @robrix , I have just checked if we can get rid of it. Seems not, so leaving it as is.